spec/models/user_spec.rb

```diff
@@ -743,4 +743,48 @@
       end
     end
   end
+
+  describe '#clear_progress_for!' do
+    let(:user) { create :user }
+
+    let(:organization_1) { create :organization, name: 'orga-1' }
+    let(:organization_2) { create :organization, name: 'orga-2' }
+    let(:exercise_1) { create :indexed_exercise }
+    let(:exercise_2) { create :indexed_exercise }
+
+    let(:messages_count) { Message.count }
+
+    before do
+      organization_1.switch!
+      assignment_1 = exercise_1.submit_solution!(user).tap(&:passed!)
+      create :message, sender: user, assignment: assignment_1, discussion_id: nil
+      organization_2.switch!
+      assignment_2 = exercise_2.submit_solution!(user).tap(&:passed!)
+      create :message, sender: user, assignment: assignment_2, discussion_id: nil
+      create :message, sender: user, assignment: assignment_1, discussion_id: 1
+    end
```

Comment:
I don't think this last message here can actually occur. Discussion messages don't have an assignment, as far as I know; only teacher-to-student messages do (but then they don't have a discussion id).

In any case the behavior is the same, I think.

Reply:
ah, I was not aware of that. I'll do a quick check on it later just in case

Reply:
yep, you were absolutely right! as a side note we have 2 messages which have both `assignment: nil` and `discussion_id: nil`... we should probably delete those

Reply:
~We should delete all messages with `discussion:nil` and the whole feature itself~ Yeah, we should